---
title: "bioinfo_lab4"
author: "Milda Poceviciute"
date: "6 December 2018"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(GEOquery)
library(simpleaffy)
library(RColorBrewer)
library(affyPLM)
library(limma)
library(hgu133plus2.db)
library(annotate)
library(ggplot2)
```


## Question 1

At first they import the data file of the Comparative Gene Expression Profiling of HUVEC and Ocular Vascular Endothelial Cells (Homo Sapiens) as a dataframe _phenodata_. They also include which target variables for which data set they will attempt to model.

```{r, eval=FALSE}
## ---- cache=TRUE---------------------------------------------------------
x = getGEOSuppFiles("GSE20986")
x
## ------------------------------------------------------------------------
untar("GSE20986/GSE20986_RAW.tar", exdir = "data")

## ------------------------------------------------------------------------
cels = list.files("data/", pattern = "[gz]")
sapply(paste("data", cels, sep = "/"), gunzip)

## ------------------------------------------------------------------------
phenodata = matrix(rep(list.files("data"), 2), ncol =2)
class(phenodata)
phenodata <- as.data.frame(phenodata)
colnames(phenodata) <- c("Name", "FileName")
phenodata$Targets <- c("iris", 
                       "retina", 
                       "retina", 
                       "iris", 
                       "retina", 
                       "iris", 
                       "choroid", 
                       "choroid", 
                       "choroid", 
                       "huvec", 
                       "huvec", 
                       "huvec")
write.table(phenodata, "data/phenodata.txt", quote = F, sep = "\t", row.names = F)
```

Then the data is read again as AffyBatch object into variable _celfiles_, the data comes from a MicroArray of a company Affymetrix. Afterwards the box plots are produced which show the statistics of the 12 samples (minimum, first quartile, median, third quartile, and maximum). One of them is coloured by the samples.

```{r}
library(simpleaffy)
library(affy)
celfiles <- read.affy(covdesc = "phenodata.txt", path = "data")
boxplot(celfiles)
library(RColorBrewer)
cols = brewer.pal(8, "Set1")
eset <- affy::exprs(celfiles)
samples <- celfiles$Targets
colnames(eset)
colnames(eset) <- samples

boxplot(celfiles, col = cols, las = 2)

```


Then the distance matrix is created for the genome data from 12 samples, and the _hclus_ function is used to create cluster dendogram of the samples:

```{r}
distance <- dist(t(eset), method = "maximum")
clusters <- hclust(distance)
plot(clusters)
```

The genomic data is normalised and converted into ExpressionSet format, the bar plots of normalised and non-normalised data are plotted for comparison purposes:

```{r}
library(simpleaffy)
library(affyPLM)
celfiles.gcrma = gcrma(celfiles)
par(mfrow=c(1,2))
boxplot(celfiles.gcrma, col = cols, las = 2, main = "Post-Normalization");
boxplot(celfiles, col = cols, las = 2, main = "Pre-Normalization")
```


And then the clusturing is performed on a normalise data set:

```{r}
distance2 <- dist(t(affy::exprs(celfiles.gcrma)), method = "maximum")
clusters2 <- hclust(distance2)
plot(clusters2)

```

Here the design matrix is created which indicates what explanatory variables are present in each  of the 12 samples. This matrix is used to fit the linear regression model. Also a contrast matrix is created. It has three pairs of the explanatory variables, and each row indicates whether a single explanatory variable is present or not in each of the pairs. This is used in fitting  a microarray linear model. The result shows the resulting statistics for the top 100000 genes for huvec_choroid variable. 


```{r}
library(limma)


samples <- as.factor(samples)
design <- model.matrix(~0+samples)
colnames(design)
colnames(design) <- c("choroid", "huvec", "iris", "retina")
design
contrast.matrix = makeContrasts(
    huvec_choroid = huvec - choroid, 
    huvec_retina = huvec - retina, 
    huvec_iris = huvec - iris, 
    levels = design)

fit = lmFit(celfiles.gcrma, design)
huvec_fit <- contrasts.fit(fit, contrast.matrix)
huvec_ebay <- eBayes(huvec_fit)

library(hgu133plus2.db)
library(annotate)
probenames.list <- rownames(topTable(huvec_ebay, number = 100000))
getsymbols <- getSYMBOL(probenames.list, "hgu133plus2")
results <- topTable(huvec_ebay, number = 100000, coef = "huvec_choroid")
results <- cbind(results, getsymbols)
```



Here the summary of the results are printed out, and the genes are grouped into three groups (a new variable "threshold" is introduced for that). Threshold is 1 if p-value is above $\alpha = 0.05$ significant level, the genes that have p-values below $\alpha = 0.05$ and logFC >5, and those that have p-values below $\alpha = 0.05$, and logFC >-5. logFC is a log2-fold-change corresponding to a contrast.


```{r}
summary(results)

results$threshold <- "1"
a <- subset(results, adj.P.Val < 0.05 & logFC > 5)
results[rownames(a), "threshold"] <- "2"
b <- subset(results, adj.P.Val < 0.05 & logFC < -5)
results[rownames(b), "threshold"] <- "3"
table(results$threshold)

```

The plot below illustrates the results. The downregulated genes are created by the process when a cell decreases the quantity of a cellular component (such as RNA or protein) in response to an external stimulus. The upregulated genes are greated by a reversed process.

```{r}
library(ggplot2)
volcano <- ggplot(data = results, 
                  aes(x = logFC, y = -1*log10(adj.P.Val), 
                      colour = threshold, 
                      label = getsymbols))

volcano <- volcano + 
    geom_point() + 
    scale_color_manual(values = c("black", "red", "green"), 
                       labels = c("Not Significant", "Upregulated", "Downregulated"), 
                       name = "Key/Legend")

volcano + 
    geom_text(data = subset(results, logFC > 5 & -1*log10(adj.P.Val) > 5), aes(x = logFC, y = -1*log10(adj.P.Val), colour = threshold, label = getsymbols)  )

```



## Question 2

Contrast matrices for the remaining pairs (DO WE NEED TO DO THIS???)

```{r}
contrast.matrix = makeContrasts(
    huvec_choroid = huvec - choroid, 
    huvec_retina = huvec - retina, 
    huvec_iris = huvec - iris, 
    levels = design)

```


```{r}
# Not normalised data
data <- eset[,c(1,2,7,11)]
colnames(data) <- c("iris","retina","choroid","huvec")
plot(x=data[,1],y=data[,2],xlab="iris",ylab="retina")
plot(x=data[,1],y=data[,3],xlab="iris",ylab="choroid")
plot(x=data[,1],y=data[,4],xlab="iris",ylab="huvec")
plot(x=data[,2],y=data[,3],xlab="retina",ylab="choroid")
plot(x=data[,2],y=data[,4],xlab="retina",ylab="huvec")
plot(x=data[,3],y=data[,4],xlab="choroid",ylab="huvec")
# Normalised data
comparison1 <- pairwise.comparison(celfiles.gcrma,"Targets",c("iris","retina"))
comparison2 <- pairwise.comparison(celfiles.gcrma,"Targets",c("iris","choroid"))
comparison3 <- pairwise.comparison(celfiles.gcrma,"Targets",c("iris","huvec"))
comparison4 <- pairwise.comparison(celfiles.gcrma,"Targets",c("retina","huvec"))
comparison5 <- pairwise.comparison(celfiles.gcrma,"Targets",c("retina","choroid"))
comparison6 <- pairwise.comparison(celfiles.gcrma,"Targets",c("choroid","huvec"))

plot(comparison1)
plot(comparison2)
plot(comparison3)
plot(comparison4)
plot(comparison5)
plot(comparison6)
```


The dendograms and heatmaps are produced for both: raw data cluster analysis and the normalised data cluster analysis.

```{r}
library("RColorBrewer")
library("gplots")
plot(clusters)
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
dist_mat <- as.matrix(distance)
# Heatmap of the raw data
heatmap.2( dist2_mat, Rowv=as.dendrogram(clusters),
          symm=TRUE, trace="none", col=colors,
          margins=c(2,10), labCol=FALSE )
plot(clusters2)
# Heatmap of the normalised data
dist2_mat <- as.matrix(distance2)
heatmap.2( dist2_mat, Rowv=as.dendrogram(clusters2),
          symm=TRUE, trace="none", col=colors,
          margins=c(2,10), labCol=FALSE )
```


## QUestion 3

The genes are grouped into three groups that contains genes that have:
*   p-value above $\alpha = 0.05$ significant level (insignificant)
*   p-values below $\alpha = 0.05$ and logFC >5 (Upregulated)
*   p-values below $\alpha = 0.05$, and logFC >-5 (Downregulated)

logFC is a log2-fold-change corresponding to a contrast.

```{r}
## huvec-retina
results2 <- topTable(huvec_ebay, number = 100000, coef = "huvec_retina")
results2 <- cbind(results2, getsymbols)

results2$threshold <- "1"
a <- subset(results2, adj.P.Val < 0.05 & logFC > 5)
results2[rownames(a), "threshold"] <- "2"
b <- subset(results2, adj.P.Val < 0.05 & logFC < -5)
results2[rownames(b), "threshold"] <- "3"
table(results2$threshold)

## huvec-iris
results3 <- topTable(huvec_ebay, number = 100000, coef = "huvec_iris")
results3 <- cbind(results3, getsymbols)

results3$threshold <- "1"
a <- subset(results3, adj.P.Val < 0.05 & logFC > 5)
results3[rownames(a), "threshold"] <- "2"
b <- subset(results3, adj.P.Val < 0.05 & logFC < -5)
results3[rownames(b), "threshold"] <- "3"
table(results3$threshold)
```

Vulcano plot for the huvec-retina pair:

```{r}
library(ggplot2)
volcano <- ggplot(data = results2, 
                  aes(x = logFC, y = -1*log10(adj.P.Val), 
                      colour = threshold, 
                      label = getsymbols))

volcano <- volcano + 
    geom_point() + 
    scale_color_manual(values = c("black", "red", "green"), 
                       labels = c("Not Significant", "Upregulated", "Downregulated"), 
                       name = "Key/Legend")

volcano + 
    geom_text(data = subset(results2, logFC > 5 & -1*log10(adj.P.Val) > 5), aes(x = logFC, y = -1*log10(adj.P.Val), colour = threshold, label = getsymbols)  )

```


Vulcano plot for the huvec-iris pair:

```{r}
volcano <- ggplot(data = results3, 
                  aes(x = logFC, y = -1*log10(adj.P.Val), 
                      colour = threshold, 
                      label = getsymbols))

volcano <- volcano + 
    geom_point() + 
    scale_color_manual(values = c("black", "red", "green"), 
                       labels = c("Not Significant", "Upregulated", "Downregulated"), 
                       name = "Key/Legend")

volcano + 
    geom_text(data = subset(results3, logFC > 5 & -1*log10(adj.P.Val) > 5), aes(x = logFC, y = -1*log10(adj.P.Val), colour = threshold, label = getsymbols)  )

```


## Question 4


```{r}
# significant genes

sig1 <- unique (subset(results, logFC > 5 & -1*log10(adj.P.Val) > 5)$getsymbols)
sig2 <- unique (subset(results2, logFC > 5 & -1*log10(adj.P.Val) > 5)$getsymbols)
sig3 <- unique (subset(results3, logFC > 5 & -1*log10(adj.P.Val) > 5)$getsymbols)
# all_significant = unique(c( as.character(sig1), as.character(sig2), as.character(sig3)))
common_genes <- intersect(intersect(sig1,sig2),sig3)
cat("The significant genes that have a substential effect are: ")
as.character(common_genes)[2:length(common_genes)]
```

*   HOXB7: 
The HOXB7 gene encodes a protein with a homeobox DNA-binding domain. It is a member of the Antp homeobox family and is included in a cluster of homeobox B genes that are located on chromosome 17. The encoded protein works as a transcription factor and is involved in cell proliferation and differentiation. Increased expression of HOXB7 has been associated with cases of melanoma and ovarian cancer. The gene is expressed in several tissues, including (but not limited to) kidney, colon, small intestine, urinary bladder, and fat tissue.

https://www.ncbi.nlm.nih.gov/gene/3217

*   IL1RL1:
This gene, that is found on human Chromosome 2, is enconding a protein thta is a member of the interleukin 1 receptor family. The gene itself is part of the Toll-like receptors superfamily which is known to play an important role in the human innate immune system. Mutations in this gene have been linked to atopic dermatitis and asthma, while it is also involved in the progression of some cardiac diseases. The protein encoded by this gene is an indicator of the severity or presence of cardiatic diseases. The gene is expressed in placenta, kidney, lung, gal bladder and adrenal tissues.


*   HOXA5:


HOXA5 is a gene that is part of the A cluster on chromosome 7 and encodes the Homeobox Hox-A5 protein. It encodes a DNA-binding transcription factor that regulates gene expression, morphogenesis, and differentiation. Since the protein that HOXA5 encodes upregulates a tumor supressor, the protein may be important for tumor formation. HoxA5 is suppressed in acute myeloid leukemia. The gene is expressed in several tissues, for example, adrenal, lung, kidney, and fat.

https://en.wikipedia.org/wiki/HOXA5

*Gene Ontology terms associated with each gene:*

*HOXB7:*

_Molecular function_
* DNA binding
* sequence-specific DNA binding
* transcription factor activity, sequence-specific DNA binding
* transcriptional activator activity, RNA polymerase II core promoter proximal region sequence-specific binding
* RNA polymerase II core promoter proximal region sequence-specific DNA binding
* protein binding
* RNA polymerase II transcription factor activity, sequence-specific DNA binding
* RNA polymerase II distal enhancer sequence-specific DNA binding
* transcription factor activity, RNA polymerase II distal enhancer sequence-specific binding
* transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific binding
_Cellular component_	
* cell nucleus
_Biological process_	
* embryonic skeletal system morphogenesis
* pattern specification process
* skeletal system development
* respiratory system process
* regulation of transcription, DNA-templated
* multicellular organism growth
* lung development
* trachea morphogenesis
* positive regulation of receptor biosynthetic process
* bronchiole development
* cell-cell signaling involved in mammary gland development
* regulation of mammary gland epithelial cell proliferation
* mammary gland alveolus development
* lung goblet cell differentiation
* transcription, DNA-templated
* mammary gland epithelial cell differentiation
* negative regulation of erythrocyte differentiation
* respiratory gaseous exchange
* trachea cartilage morphogenesis
* embryonic skeletal system development
* morphogenesis of an epithelium
* multicellular organism development
* thyroid gland development
* lobar bronchus epithelium development
* mesenchymal-epithelial cell signaling
* positive regulation of apoptotic process
* intestinal epithelial cell maturation
* negative regulation of angiogenesis
* cartilage morphogenesis
* lung alveolus development
* lung-associated mesenchyme development
* epithelial tube branching involved in lung morphogenesis
* cell migration
* positive regulation of myeloid cell differentiation
* anterior/posterior pattern specification
* positive regulation of transcription from RNA polymerase II promoter
* transcription from RNA polymerase II promoter

source (https://en*wikipedia*org/wiki/HOXA5)

*IL1RL1:* 

_Molecular function_	
* interleukin-33 receptor activity
* interleukin-1 receptor activity
* cytokine receptor activity
* protein binding
* signal transducer activity, downstream of receptor
* interleukin-33 binding
_Cellular component_	
* integral component of membrane
* external side of plasma membrane
* membrane
* extracellular region
* plasma membrane
* extracellular matrix
_Biological process_	
* negative regulation of T-helper 1 type immune response
* positive regulation of inflammatory response
* negative regulation of interferon-gamma production
* positive regulation of macrophage activation
* negative regulation of I-kappaB kinase/NF-kappaB signaling
* positive regulation of chemokine secretion
* signal transduction
* immune response
* positive regulation of interleukin-5 production
* interleukin-33-mediated signaling pathway

source (https://en.wikipedia.org/wiki/IL1RL1)

*HOXA5:*

_Molecular function_	
* DNA binding
* sequence-specific DNA binding
* transcription factor activity, sequence-specific DNA binding
* transcriptional activator activity, RNA polymerase II core promoter proximal region sequence-specific binding
* RNA polymerase II core promoter proximal region sequence-specific DNA binding
* protein binding
* RNA polymerase II transcription factor activity, sequence-specific DNA binding
* RNA polymerase II distal enhancer sequence-specific DNA binding
* transcription factor activity, RNA polymerase II distal enhancer sequence-specific binding
* transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific binding
_Cellular component_	
* cell nucleus
_Biological process_	
* embryonic skeletal system morphogenesis
* pattern specification process
* skeletal system development
* respiratory system process
* regulation of transcription, DNA-templated
* multicellular organism growth
* lung development
* trachea morphogenesis
* positive regulation of receptor biosynthetic process
* bronchiole development
* cell-cell signaling involved in mammary gland development
* regulation of mammary gland epithelial cell proliferation
* mammary gland alveolus development
* lung goblet cell differentiation
* transcription, DNA-templated
* mammary gland epithelial cell differentiation
* negative regulation of erythrocyte differentiation
* respiratory gaseous exchange
* trachea cartilage morphogenesis
* embryonic skeletal system development
* morphogenesis of an epithelium
* multicellular organism development
* thyroid gland development
* lobar bronchus epithelium development
* mesenchymal-epithelial cell signaling
* positive regulation of apoptotic process
* intestinal epithelial cell maturation
* negative regulation of angiogenesis
* cartilage morphogenesis
* lung alveolus development
* lung-associated mesenchyme development
* epithelial tube branching involved in lung morphogenesis
* cell migration
* positive regulation of myeloid cell differentiation
* anterior/posterior pattern specification
* positive regulation of transcription from RNA polymerase II promoter
* transcription from RNA polymerase II promoter

As we can see from the lists of the GO above, all three genes are involved in protein binding. Protein binding is the ability of the proteins to form bonds with other substances, commonly this refers to the drugs binding to the proteins in blood. Hence, all of them play a role into how drugs and human bodies interact (they can enhance or detract the drug performances).



